src/gui/remote/DatabaseSettingsWidgetRemote.cpp

```diff
@@ -191,10 +192,24 @@ void DatabaseSettingsWidgetRemote::testDownload()
     }
 
     if (!QFile::exists(result.filePath)) {
-        m_ui->messageWidget->showMessage(tr("Download finished, but file %1 could not be found.").arg(result.filePath),
+        m_ui->messageWidget->showMessage(tr("Command finished, but downloaded file does not exist."),
+                                         MessageWidget::Error);
+        return;
+    }
+
+    if (!hasValidPublicHeaders(result.filePath)) {
+        m_ui->messageWidget->showMessage(tr("Download finished, but file failed KeePass header check. File is not a "
+                                            "KeePass file or it's an unsupported version"),
                                          MessageWidget::Error);
         return;
     }
 
     m_ui->messageWidget->showMessage(tr("Download successful."), MessageWidget::Positive);
+}
+
+bool DatabaseSettingsWidgetRemote::hasValidPublicHeaders(QString& filePath) {
+    // Read public headers
+    QString error;
+    QScopedPointer<Database> db(new Database());
+    return db->open(filePath, nullptr, &error);
```

Comment:
You could inline this since you have only a single call for it and use the error string.

Reply:
I wasn't sure if I should use the error string, as it can be quite long and may not fit in the message box. I added the error anyway, as it is better to have it rather than leave the user in the dark.

I did not inline the method though, as I prefer having a named method, rather than calling `Database::open` with a `nullptr` as second parameter, which implies reading the headers.

Reply:
Since we are repeating this pattern we should introduce a `Database::readHeaders` function that takes care of the particulars.